src/init.cpp

```diff
@@ -1256,10 +1256,12 @@ bool AppInitMain(NodeContext& node)
     InitSignatureCache();
     InitScriptExecutionCache();
 
-    LogPrintf("Using %u threads for script verification\n", nScriptCheckThreads);
     if (nScriptCheckThreads) {
+        LogPrintf("Using %u threads concurrently for script verification\n", nScriptCheckThreads);
         for (int i=0; i<nScriptCheckThreads-1; i++)
             threadGroup.create_thread([i]() { return ThreadScriptCheck(i); });
+    } else {
+        LogPrintf("Using the only thread for script verification\n");
```

Comment:
```suggestion
        LogPrintf("Script verification is done in the main thread\n");
```

Reply:
But script verification is always done in the main thread, the other threads are only additional :smile:

Reply:
"Script verification is done in the main thread only (no validation concurrency)"

Reply:
or "Script verification uses N additional threads" where N can be 0…

Reply:
> Yeah, let's make it a one-line "fix"

Not sure about "one-line fix" as actual _additional_ threads number is `nScriptCheckThreads-1`.

Reply:
True, it is shorter than conditional operator ;)

Reply:
I don't think this should be `nScriptCheckThreads-1`. `nScriptCheckThreads` is the number of threads that are dedicated to script checking, so:

- if `nScriptCheckThreads` >= 2, then log "Script verification uses {nScriptCheckThreads} additional threads"
- if `nScriptCheckThreads` == 0, then script checking is done on the main thread so log "Script verification uses 0 additional threads"
- `nScriptCheckThreads` cannot == 1 because of the logic in init.cpp

so you just want:

`LogPrintf("Script verification uses %d dedicated threads\m", nScriptCheckThreads);`

The log is already essentially correct. I think you just need to add the word 'dedicated' or 'additional' (I prefer 'dedicated').

Reply:
That's not correct; it should be `LogPrintf("Script verification uses %d dedicated threads\m", nScriptCheckThreads - 1);` then.

Reply:
@sipa The `std::max(nScriptCheckThreads-1, 0)` bound is there because nScriptCheckThreads might be zero at this point.